### PR TITLE
move crypto provider installer to be called once

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
this PR must be followed by an addition to zingo-mobile rust to also call this installer on app launch

### CODE REVIEW

**Comment by juanky201271** [comment]
I added the FFI function in zingo-mobile:
https://github.com/zingolabs/zingo-mobile/pull/696/commits/62d68d765e948e1810af02f25d79b3ad980404bf
And it's working, no run-time error about the Crypto Provider.

**Comment by fluidvanadium** [comment]
i am proud of these sync tests. i understand that they have variable run time, but i do not understand why they would be ignored

**Review by nachog00 — COMMENTED** [review]
Whats our stance on expects and unwraps? Do we allow them in tests only?
@zancas @Oscar-Pepper

**Comment by Oscar-Pepper** [comment]
> i am proud of these sync tests. i understand that they have variable run time, but i do not understand why they would be ignored

as you said, we cant have tests unignored that are guaranteed to slow down test run time, they will just get ignored some time in the near future  anyway and cost a load of CI / local test developer time. this is what darkside is for, we have chain load framework. the only limitation I am aware of is transparent

**Comment by Oscar-Pepper** [comment]
> Whats our stance on expects and unwraps? Do we allow them in tests only? @zancas @Oscar-Pepper

unwraps are test-only. expects can be acceptable in production with a message that eludes to why it should NEVER panic.

**Comment by zancas** [comment]
In order to guarantee non-panic on an `expect` in production, we'd need to add wrapping correctness guarantees.   That's not a proper effort for a developer unless they're already **expert** in the code base.

Please do **NOT** add `expect`s to production code.

**Comment by zancas** [comment]
> > i am proud of these sync tests. i understand that they have variable run time, but i do not understand why they would be ignored
> 
> as you said, we cant have tests unignored that are guaranteed to slow down test run time, they will just get ignored some time in the near future anyway and cost a load of CI / local test developer time. this is what darkside is for, we have chain load framework. the only limitation I am aware of is transparent

I disagree.   This is clear evidence that we do not have sufficient resources allocated to CI.   We should *not* degrade our test quality because the free tier of github runners is too slow.   That's the wrong direction.   We need to invest in test infrastructure.

File: zingocli/src/main.rs [hunk 1]
```diff
@@ -1,4 +1,8 @@
 #![forbid(unsafe_code)]
 pub fn main() {
+    // install default crypto provider (ring)
+    if let Err(e) = rustls::crypto::ring::default_provider().install_default() {
+        eprintln!("Error installing crypto provider: {:?}", e)
```
Comment (sebastianlujan) [inline zingocli/src/main.rs:5]: You could use `std::process:exit(1) ` to exit if the default crypto provider fails.

File: libtonode-tests/tests/concrete.rs [hunk 2]
```diff
@@ -3452,7 +3452,7 @@ mod slow {
             let chainwait: u32 = 6;
             let amount: u64 = u64::from(chainwait * i);
             zingolib::testutils::increase_server_height(&regtest_manager, chainwait).await;
-            let _ = recipient.do_sync(false).await;
+            recipient.do_sync(false).await.unwrap();
```
Comment (sebastianlujan) [inline libtonode-tests/tests/concrete.rs:3455]: Unwap is generally discouraged in production, because could generate panics.

Comment (zancas) [reply libtonode-tests/tests/concrete.rs:3455]: This is a good point!  And you're correct for production code in `lib` crates.   However we generally permit `unwrap` in test code, so it's ok in the `libtonode-tests` crate.

File: zingocli/src/main.rs (hunk 1 above)
Comment (zancas) [reply zingocli/src/main.rs:5]: This is an interesting option.   What are the trade-offs between these approaches?

File: zingolib/src/lightclient/sync.rs [hunk 3]
```diff
@@ -681,11 +681,12 @@ pub mod test {
         std::env::set_var("RUST_BACKTRACE", "1");
         let wallet = wallet_case.load_example_wallet().await;
         let lc = LightClient::create_from_wallet_async(wallet).await.unwrap();
-        let _ = lc.do_sync(true).await;
+        lc.do_sync(true).await.unwrap();
         lc
     }
 
     /// this is a live sync test. its execution time scales linearly since last updated
+    #[ignore = "testnet and mainnet tests should be ignored due to increasingly large execution times"]
```
Comment (zancas) [inline zingolib/src/lightclient/sync.rs:689]: I am OK with this as a StopGap while we invest in test infrastructure.

File: zingolib/src/lightclient/sync.rs [hunk 4]
```diff
@@ -681,11 +681,12 @@ pub mod test {
         std::env::set_var("RUST_BACKTRACE", "1");
         let wallet = wallet_case.load_example_wallet().await;
         let lc = LightClient::create_from_wallet_async(wallet).await.unwrap();
-        let _ = lc.do_sync(true).await;
+        lc.do_sync(true).await.unwrap();
```
Comment (zancas) [inline zingolib/src/lightclient/sync.rs:684]: May as well make it an:

`expect("If sync panics here, then the test is already interesting.")`

Or perhaps there's a better string.   I don't see why we wouldn't use expect.

**Comment by zancas** [comment]
Alright  I added some non-blocking comments.

**Review by zancas — APPROVED** [review]
Given @juanky201271 's test, and caveats about upgrading infrastructure..  I approve.

**Pull request merged**